AWSPinpoint/AWSPinpointTargetingClient.m

```diff
@@ -207,8 +275,14 @@ - (void)addAttribute:(NSArray *)theValue
     @synchronized(self) {
         //Save value to disk
         [self.globalAttributes setValue:theValue forKey:theKey];
-        [self.context.configuration.userDefaults setObject:self.globalAttributes forKey:AWSPinpointEndpointAttributesKey];
-        [self.context.configuration.userDefaults synchronize];
+        NSError *codingError;
+        NSData *globalAttributesData = [AWSNSCodingUtilities versionSafeArchivedDataWithRootObject:self.globalAttributes
+                                                                            requiringSecureCoding:YES
+                                                                                            error:&codingError];
+        if (codingError) {
+            AWSDDLogError(@"Error archiving globalAttributesData. Updating service but not persisting locally: %@", codingError);
```

Comment:
This is still persisting even though there's an error

Reply:
Missed it, should be fixed now.